fmriprep/interfaces/multiecho.py

```diff
@@ -78,320 +74,44 @@ def _run_interface(self, runtime):
-            with at least two echos
-
-    """
-    # First, load each echo and average over time
-    echos = [np.mean(nb.load(e).get_data(), axis=-1) for e in echo_list]
-
-    # In the first echo, find the 33rd percentile and the voxel(s)
-    # whose average activity is equal to that value
-    perc33 = np.percentile(echos[0][echos[0].nonzero()],
-                           33, interpolation="higher")
-    med_vox = (echos[0] == perc33)
-
-    # For each (averaged) echo, extract the max signal in the
-    # identified voxels and divide it by 3-- save as a threshold
-    thrs = np.hstack([np.max(echo[med_vox]) / 3 for echo in echos])
-
-    # Let's stack the arrays to make this next bit easier
-    emeans = np.stack(echos, axis=-1)
-
-    # Now, we want to find all voxels (in each echo) that show
-    # absolute signal greater than our echo-specific threshold
-    mthr = np.ones_like(emeans)
-    mthr *= thrs[np.newaxis, np.newaxis, np.newaxis, :]
-    voxels = np.abs(emeans) > mthr
-
-    # We save those voxel indices out to an array
-    last_emask = np.array(voxels, dtype=np.int).sum(-1)
-    # Save a mask of voxels sampled by at least two echos
-    two_emask = (last_emask != 0)
-
-    return last_emask, two_emask
-
-
-def define_t2s_map(echo_list, tes, last_emask, two_emask):
-    """
-    Computes the quantiative T2* mapping according to
-    :math:`ΔS/S = ΔS0/S0 − ΔR2 * TE`.
-
-    **Inputs**
-
-        echo_list
-            list of file names for all echos
-        tes
-            echo times for the multi-echo EPI run
-        last_emask
-            numpy array where voxel values correspond to which
-            echo a voxel can last be sampled with
-        two_emask
-            boolean array of voxels that can be sampled
-            with at least two echos
-
-    **Outputs**
-
-        t2s_map
-            the T2* map for the EPI run
-    """
-    # get some basic shape information
-    echo_stack = np.stack([nb.load(echo).get_data() for echo in echo_list],
-                          axis=-2)
-    nx, ny, nz, necho, nt = echo_stack.shape
-
-    # create empty arrays to fill later
-    t2ss = np.zeros([nx, ny, nz, necho - 1])
-    s0vs = t2ss.copy()
-
-    # consider only those voxels sampled by at least two echos
-    two_edata = echo_stack[two_emask.nonzero()]
-    two_echo_nvox = two_edata.shape[0]
-
-    # for the second echo on, do log linear fit
-    for echo in range(2, necho + 1):
+        # tedana expects echo times in milliseconds, rather than seconds
+        echo_times = [te * 1000 for te in self.inputs.te_list]
 
-        # multiply by 1000, so in ms rather than s
-        # ΔS/S = ΔS0/S0 − ΔR2 * TE, so take neg TEs
-        neg_tes = [-1000 * te for te in tes[:echo]]
+        (self._results['t2sv'], self._results['s0v'],
+         self._results['t2svG'], self._results['s0vG'],
+         self._results['ts_OC']) = tedana.t2smap_workflow(self.inputs.in_files, echo_times)
```

Comment:
I'm wondering if the issue is here. [`t2smap_workflow`](https://github.com/ME-ICA/tedana/blob/e7f04ff299412befbf309ceece17cd6e4889038e/tedana/workflows/t2smap.py#L78-L194) does not seem to return anything.

I'm not sure why this would result in a failure like the issue you linked to, but sometimes things cascade weirdly.

Reply:
You're totally right -- it writes out the files rather than returning a value. I'll go back to some of the other nipype interfaces and check to see how they're capturing outputs.